### DISCUSSION TITLE
I have a few newbie questions that I would appreciate if you could clarify.

### DISCUSSION THREAD

**Discussion by ember11498** [sai3010/Typesense-Kubernetes-Operator#27] [Q&A]
Hello,

I want to host my typesense in my kubernetes cluster so I guess I will do git clone "https://github.com/sai3010/Typesense-Kubernetes-Operator/tree/main/templates".

However, I have a few questions regarding the storage and how it works exactly.

I noticed you have a StorageClass named typesense-storage however in your statefulset you dont use it. It seems to me you are using local storage in each node when you mountvolumes in certain paths?

So if we store a lot of data in typesense isnt 256Mi memory limit very low? maybe this is the memory just to run typesense and not to store the data itself? Can you clarify where is the data being stored? Also, do I need to adjust the 256Mi memory limit if i plan to store a lot of data?

also does indexed data and normal storage data get storaged inside the node? or does any of these data get stored by a cloud provider using the storage class typesense-storage?

**Comment by ember11498** [comment]
@sai3010 I would appreciate if you could answer this please! Thanks

**Reply by sai3010** [reply]
Hi @ember11498 ,
By default, the operator does not assign a storage class, meaning data is temporarily stored within the Typesense pod itself. For persistent data storage, we recommend creating a custom storage class and configuring it with the operator.

Regarding memory and CPU limits, the default settings are on the lower end. These values should be adjusted based on your specific scale—particularly the anticipated request rate and data size—so please plan resources accordingly.

None of the data goes to the cloud provider, it is either local to your pod or to the storage class you specify. 
Hope this answers your questions.
Please join the community if you have more questions. Happy to help there!

**Reply by ember11498** [reply]
@sai3010 thanks for your answer! just a quick follow up question.

if i use storageclass in my case it will create a volume in digital ocean automatically because my kubernetes are hosted by digital ocean. Does typesense store all data there even the indexed data? or the indexed data would always be stored locally inside the pod itself so the search is quicker?

I 100% want persistant data just thinking about the optimal way.

so your storage class is just an example of how to create it but if we want to use it we would have to still do it

**Reply by sai3010** [reply]
@ember11498 Yes if you create a storage class, a volume is assigned and every data is stored in that. I'm not sure how Typesense does things internally but overall you won't loose data if you use the storage class. But please make sure to take backup of them.

**Reply by ramonmalcolm10** [reply]
This seems each pod will get it own volume which would lead to pods not having all the data, can correct me if am wrong @sai3010.  @ember11498 did you get it to work with each pods reading and writing from one specific volume?

**Reply by sai3010** [reply]
@ramonmalcolm10, **Typesense** is deployed as a statefulset which means that each pod will have its volume, that being said all the nodes of typesense will have up to date data as they are replica sets.
I don't think its a great idea to mount 1 volume for all pods and I guess Typesense pods may not start if given a common volume.
Please join the community for more support:  https://join.slack.com/t/typesensekube-uma7122/shared_invite/zt-2tph49vef-PWT6dvXFyljMzCbV6Peq~A

**Reply by ember11498** [reply]
@sai3010 so from reading this it seems typesense works similarly to postgres streaming replication? where each postgres replica has its own volume and they all get writes from the same instance maybe?
